packages/node_modules/@webex/plugin-meetings/src/meeting-info/request.js

```diff
@@ -81,42 +11,14 @@ export default class MeetingInfoRequest extends StatelessSparkPlugin {
 
   /**
    *
-   * @param {String} destination One of many types of destinations. Currently only sip uri works
+   * @param {Object} options
    * @returns {Promise} returns a promise that resolves/rejects the result of the request
    */
-  fetchMeetingInfo(destination) {
-    let value = destination;
-    let type = SIP_URI;
-
-    // Making sure all the webEx meeting are converted into SipUrl
-    const sipUrl = checkforMeetingLink(value);
-
-    if (sipUrl) {
-      type = SIP_URI;
-      value = sipUrl;
+  fetchMeetingInfo(options) {
+    if (!options || !options.type || !options.destination) {
+      throw new Error('MeetingInfo should be fetched with a type and destination specified, see list of valid types and their corresponding values in constants');
     }
-
-    const resourceOptions = getResourceUrl(type, value);
-
-    const requestParams = {
-      method: resourceOptions.method,
-      api: LOCUS,
-      resource: resourceOptions.resource
-    };
-
-    if (resourceOptions.method === GET) {
-      // for handling URL redirections
-      requestParams.resource = requestParams.resource.concat(`&${ALTERNATE_REDIRECT_TRUE}`);
-    }
-    else if (type !== LOCUS_ID) {
-      // locus id check is a PUT not sure why
-      requestParams.body = {
-        value,
-        lookupType: type,
-        alternateRedirect: true
-      };
-    }
-
-    return this.request(requestParams);
+    const typeValue = MeetingInfoUtil.getMeetingInfoTypeValue(options);
+    return this.request(MeetingInfoUtil.getRequestParams(MeetingInfoUtil.getResourceUrl(typeValue.type, typeValue.value), typeValue.type, typeValue.value, LOCUS));
```

Comment:
Might just be me, but this is pretty hard to read. Also, can we explicitly pass the params we need from options? Or will it grow to a various combination of things?

Reply:
ill separate it line by line
the second one